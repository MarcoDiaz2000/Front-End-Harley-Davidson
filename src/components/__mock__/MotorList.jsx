import React from 'react';

const MotorList = () => (
  <div className=" flex flex-col bg-bodyBg">
    <div className="flex flex-col justify-center align-center items-center pb-8 font-roboto">
      <h1 className="font-bold text-white text-2xl">LATEST MODEL</h1>
      <h1 className="text-sm text-gray-500">please select a model</h1>
    </div>
    <div>

      <div className="w-full h-96 card">
        <div className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2">
          <img className="w-80 h-auto" src="src/img" alt="motor" />
        </div>
        <h1 className="font-bold text-white bg-blur">Model 1</h1>
        <a
          href="/motors/1"
          className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        >
          See Model
        </a>
      </div>

      <div className="w-full h-96 card">
        <div className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2">
          <img className="w-80 h-auto" src="src/img" alt="motor" />
        </div>
        <h1 className="font-bold text-white bg-blur">Model 1</h1>
        <a
          href="/motors/1"
          className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        >
          See Model
        </a>
      </div>

      <div className="w-full h-96 card">
        <div className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2">
          <img className="w-80 h-auto" src="src/img" alt="motor" />
        </div>
        <h1 className="font-bold text-white bg-blur">Model 1</h1>
        <a
          href="/motors/1"
          className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        >
          See Model
        </a>
      </div>

      <div className="w-full h-96 card">
        <div className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2">
          <img className="w-80 h-auto" src="src/img" alt="motor" />
        </div>
        <h1 className="font-bold text-white bg-blur">Model 1</h1>
        <a
          href="/motors/1"
          className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        >
          See Model
        </a>
      </div>

      <div className="w-full h-96 card">
        <div className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2">
          <img className="w-80 h-auto" src="src/img" alt="motor" />
        </div>
        <h1 className="font-bold text-white bg-blur">Model 1</h1>
        <a
          href="/motors/1"
          className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        >
          See Model
        </a>
      </div>

    </div>
  </div>
);
export default MotorList;
