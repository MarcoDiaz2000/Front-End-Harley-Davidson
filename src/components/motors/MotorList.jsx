import React from 'react';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import Slider from 'react-slick';
import PropTypes from 'prop-types';
import MotorItem from './MotorItem';

const MotorList = ({ motorcycles }) => {
  const bikes = motorcycles;
  const settings = {
    dots: true,
    infinite: false,
    speed: 500,
    slidesToShow: 3,
    slidesToScroll: 3,
    initialSlide: 0,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 3,
          infinite: true,
          dots: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
          initialSlide: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
  };

  const {
    dots, infinite, speed, slidesToShow, slidesToScroll, initialSlide, responsive,
  } = settings;

  return (
    <div className=" flex flex-col bg-bodyBg">
      <div className="flex flex-col justify-center align-center items-center pb-8 font-roboto">
        <h1 className="font-bold text-white text-2xl">LATEST MODEL</h1>
        <h1 className="text-sm text-gray-500">please select a model</h1>
      </div>
      <Slider
        dots={dots}
        infinite={infinite}
        speed={speed}
        slidesToShow={slidesToShow}
        slidesToScroll={slidesToScroll}
        initialSlide={initialSlide}
        responsive={responsive}
        className="bodyBg"
      >
        {bikes.filter((motor) => !motor.removed).map((motor) => (
          <MotorItem key={motor.id} motor={motor} />
        ))}
      </Slider>
    </div>
  );
};
export default MotorList;

MotorList.propTypes = {
  motorcycles: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      image_url: PropTypes.string.isRequired,
    }),
  ).isRequired,
};
