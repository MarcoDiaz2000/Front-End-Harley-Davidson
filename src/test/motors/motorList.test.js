import React from 'react';
import renderer from 'react-test-renderer';
import MotorList from '../../components/__mock__/MotorList';

describe('MotorList Component', () => {
  it('should render correctly', () => {
    const component = renderer.create(<MotorList />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
