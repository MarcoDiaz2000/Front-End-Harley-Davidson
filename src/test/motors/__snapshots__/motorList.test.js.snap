// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MotorList Component should render correctly 1`] = `
<div
  className=" flex flex-col bg-bodyBg"
>
  <div
    className="flex flex-col justify-center align-center items-center pb-8 font-roboto"
  >
    <h1
      className="font-bold text-white text-2xl"
    >
      LATEST MODEL
    </h1>
    <h1
      className="text-sm text-gray-500"
    >
      please select a model
    </h1>
  </div>
  <div>
    <div
      className="w-full h-96 card"
    >
      <div
        className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2"
      >
        <img
          alt="motor"
          className="w-80 h-auto"
          src="src/img"
        />
      </div>
      <h1
        className="font-bold text-white bg-blur"
      >
        Model 1
      </h1>
      <a
        className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        href="/motors/1"
      >
        See Model
      </a>
    </div>
    <div
      className="w-full h-96 card"
    >
      <div
        className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2"
      >
        <img
          alt="motor"
          className="w-80 h-auto"
          src="src/img"
        />
      </div>
      <h1
        className="font-bold text-white bg-blur"
      >
        Model 1
      </h1>
      <a
        className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        href="/motors/1"
      >
        See Model
      </a>
    </div>
    <div
      className="w-full h-96 card"
    >
      <div
        className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2"
      >
        <img
          alt="motor"
          className="w-80 h-auto"
          src="src/img"
        />
      </div>
      <h1
        className="font-bold text-white bg-blur"
      >
        Model 1
      </h1>
      <a
        className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        href="/motors/1"
      >
        See Model
      </a>
    </div>
    <div
      className="w-full h-96 card"
    >
      <div
        className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2"
      >
        <img
          alt="motor"
          className="w-80 h-auto"
          src="src/img"
        />
      </div>
      <h1
        className="font-bold text-white bg-blur"
      >
        Model 1
      </h1>
      <a
        className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        href="/motors/1"
      >
        See Model
      </a>
    </div>
    <div
      className="w-full h-96 card"
    >
      <div
        className="w-full flex justify-center items-center h-64 bg-slate-500 p-4 m-2"
      >
        <img
          alt="motor"
          className="w-80 h-auto"
          src="src/img"
        />
      </div>
      <h1
        className="font-bold text-white bg-blur"
      >
        Model 1
      </h1>
      <a
        className="px-16 pb-2 pt-1 bg-customBg flex justify-center text-white rounded-md text-center shadow-md hover:bg-customDark"
        href="/motors/1"
      >
        See Model
      </a>
    </div>
  </div>
</div>
`;
